ambry-router/src/main/java/com/github/ambry/router/GCMCryptoService.java

```diff
@@ -366,18 +366,17 @@ private static void serializeKeyRecord(ByteBuffer outputBuffer, byte[] key, Stri
       outputBuffer.putShort(KEY_RECORD_VERSION_V_1);
       outputBuffer.putInt(key.length);
       outputBuffer.put(key);
-      Utils.serializeString(outputBuffer, keyGenAlgo, StandardCharsets.US_ASCII);
+      Utils.serializeString(outputBuffer, keyGenAlgo, StandardCharsets.UTF_8);
```

Comment:
I changed this because ` String keyAlgo = Utils.readIntString(dataStream)` in `deserializeKeyRecord()` is actually using UTF-8. (line 383)